test/powershell/Language/Scripting/Generics.Tests.ps1

```diff
@@ -0,0 +1,95 @@
+﻿Describe "Generics support" -Tags "CI" {
+    #
+    # 3 types are tested, list, stack, and dictionary.
+    #
+    # list and stack are in different assemblies, and dictionary
+    # takes more than one type parameter.
+    #
+    $listType = "system.collections.generic.list"
+    $stackType = "system.collections.generic.stack"
+    $dictionaryType = "system.collections.generic.dictionary"
+
+    It 'list[Int]' {
+        $x = [system.collections.generic.list[int]]::New()
+        $x.Add(42)
+        $x.count | Should Be 1
+    }
+
+    It 'stack[Int]' {
+        $x = [system.collections.generic.stack[int]]::New()
+        $x.Push(42)
+        $x.count | Should Be 1
+    }
+
+    It 'dictionary[string, Int]' {
+        $x = [system.collections.generic.dictionary[string, int]]::New()
+        $x.foo = 42
+        $x.foo | Should Be 42
+    }
+
+    It 'list[[Int]]' {
+        $x = [system.collections.generic.list[int]]::New()
+        $x.Add(42)
+        $x.count | Should Be 1
+    }
+
+    It 'stack[[Int]]' {
+        $x = [system.collections.generic.stack[int]]::New()
+        $x.Push(42)
+        $x.count | Should Be 1
+    }
+
+    It 'dictionary[[string], [Int]]' {
+        $x = [system.collections.generic.dictionary[string, int]]::New()
+        $x.foo = 42
+        $x.foo | Should Be 42
+    }
+
+    It 'dictionaryType[dictionary[list[int],string], stack[double]]' {
+        $x = new-object "$dictionaryType[$dictionaryType[$listType[int],string], $stackType[double]]"
+        $x.gettype().fullname | should match "double"        
+    }
+
+    It 'non-generic EventHandler' {
+        # EventHandler has a generic and a non-generic.  This code caused an exception trying to
+        # use the non-generic.
+        $x = [System.EventHandler[System.Management.Automation.PSInvocationStateChangedEventArgs]]
+
+        # The error message for a generic that doesn't meet the constraints should mention which
+        # argument failed.
+        $ex = $null
+        try {
+            [nullable[object]]
+            Throw "Exception expected, execution should not have reached here"
+        } catch {
+            $_.FullyQualifiedErrorId | Should be 'TypeNotFoundWithMessage'
+            $_ | should match "\[T\]"
+        }
+    }
+
+    It 'Array' -skip:$IsCoreCLR{
+        $x = [system.array]::ConvertAll.OverloadDefinitions
+        $x | Should match "static\s+TOutput\[\]\s+ConvertAll\[TInput,\s+TOutput\]\("
+    }
+
+   It 'Class' {
```

Comment:
fixed.

---
In reply to: [78840821](https://github.com/Powershell/Powershell/pull/2118#discussion_r78840821) [](ancestors = 78840821)